FeatureTest.md

```diff
@@ -1,11 +1,49 @@
 # Feature Test
 
-Any feature not present in [the MVP](MVP.md) will have a corresponding feature
-test. This allows polyfills to elegantly emulate a feature, and allows
-developers to fallback to another implementation if they so desire.
+The [MVP](MVP.md) allows an application to query which post-MVP features are
+supported via [`has_feature`](AstSemantics.md#feature-test). This accounts for
+the pragmatic reality that features are shipped in different orders at different
+times by different engines on the Web.
 
-Feature tests will be available from
-[WebAssembly itself](AstSemantics.md#feature-test), as well as from JavaScript.
+Since some WebAssembly features add operators and all WebAssembly code in a
+module is validated ahead-of-time, the usual JS feature detection pattern:
+```
+if (foo)
+    foo();
+else
+    alternativeToFoo();
+```
+won't work in WebAssembly (if `foo` isn't supported, `foo()` will fail to
+validate).
 
-See also [better feature testing support](FutureFeatures.md#better-feature-testing-support)
-in future features.
+Instead, applications may use one of the following strategies:
+
+1. Compile several versions of a module, each assuming different feature support
+   and use `has_feature` tests to determine which version to load.
```

Comment:
AstSemantics.md does specifically mention the invariance of `has_feature`, but I can restate here.  There is a sortof philosphical difference, though, between serving 1 file and 2.  I've seen multiple devs (e.g., Unity) say they'd prefer to serve only 1 file.